src/passes/DebugLocationPropagation.cpp

```diff
@@ -64,6 +64,10 @@ struct DebugLocationPropagation
         if (auto it = locs.find(previous); it != locs.end()) {
           locs[curr] = it->second;
         }
+      } else if (self->getFunction()->prologLocation.size()) {
+        // The first instruction may inherit its location from the
+        // function prolog
+        locs[curr] = *self->getFunction()->prologLocation.begin();
```

Comment:
How does this ensure it only affects the first instruction and not later ones?

Reply:
It should affect later instructions as well.
This piece of code:
```wasm
 ;;@ a:1:1
 (func
    ;;@ a:1:1
    (nop)
    ;;@ a:1:1
    (nop)
 )
```
is printed as:
```wasm
 ;;@ a:1:1
 (func
  (nop)
  (nop)
 )
```
So they should result in the same debug information when read back.

Reply:
Ah, you mean by the normal rules of propagating from the parent to children? So the first instruction inherits from the prolog, and children of the first instruction inherit from the first instruction, so effectively they inherit from the prologue too? If so that makes sense but the comment was confusing me. Perhaps `The first instruction may inherit its location from the function prolog` could be `Instructions may inherit their locations from the function prolog.` - ?

Reply:
Actually, on reading the code again, I think a simpler way to do this would be, instead of adding code here, to add code to `getPrevious`. That method could look at the function prologue if it does not find something else.

Reply:
I'm not sure what you mean since `getPrevious` returns a pointer to an expression. Do you mean I should move some of the code here to `getPrevious` so that it returns an optional location instead?

Reply:
But `prologLocation` has type `std::set<Function::DebugLocation>`.

Reply:
Ah, right. Then more code would need to move into `getPrevious`, and then really it would need to be renamed. I agree it's not worth it, sorry for not thinking this totally through.